File: backend/app/schemas/menu_action_permission.py
# E:\Audit\AMS\backend\app\schemas\menu_action_permission.py

from datetime import datetime
from pydantic import BaseModel, ConfigDict

class MenuActionPermissionCreate(BaseModel):
    menu_action_id: int
    permission_id: int


class MenuActionPermissionActionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    menu_id: int
    menu_key: str | None = None
    menu_title: str | None = None

    action_key: str
    action_title: str
    permission_key: str
    button_color: str | None = None
    button_icon: str | None = None
    sort_order: int
    is_active: bool
    is_visible: bool


class MenuActionPermissionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    menu_action_id: int
    permission_id: int
    is_active: bool

    menu_id: int | None = None
    menu_key: str | None = None
    menu_title: str | None = None

    action_key: str | None = None
    action_title: str | None = None
    action_permission_key: str | None = None

    permission_key: str | None = None
    resource_type: str | None = None
    resource_key: str | None = None
    action: str | None = None
    description: str | None = None

    created_by: str | None = None
    updated_by: str | None = None
    created_at: datetime
    updated_at: datetime


class MenuActionPermissionMessageResponse(BaseModel):
    message: str
    data: MenuActionPermissionResponse | None = None

File: frontend/src/types/menuActionPermission.ts
// E:\Audit\AMS\frontend\src\types\menuActionPermission.ts

export type MenuActionLookup = {
  id: number;
  menu_id: number;
  menu_key: string | null;
  menu_title: string | null;

  action_key: string;
  action_title: string;
  permission_key: string;
  button_color: string | null;
  button_icon: string | null;
  sort_order: number;
  is_active: boolean;
  is_visible: boolean;
};

export type MenuActionPermission = {
  id: number;
  menu_action_id: number;
  permission_id: number;
  is_active: boolean;

  menu_id?: number | null;
  menu_key?: string | null;
  menu_title?: string | null;

  action_key?: string | null;
  action_title?: string | null;
  action_permission_key?: string | null;

  permission_key?: string | null;
  resource_type?: string | null;
  resource_key?: string | null;
  action?: string | null;
  description?: string | null;

  created_by: string | null;
  updated_by: string | null;
  created_at: string;
  updated_at: string;
};

export type MenuActionPermissionPayload = {
  menu_action_id: number;
  permission_id: number;
};

export type MenuActionPermissionMessageResponse = {
  message: string;
  data: MenuActionPermission | null;
};